simtools/layout/telescope_position.py

```diff
@@ -17,597 +15,547 @@ class MissingInputForConvertion(Exception):
+                self._logger.error(
+                    "Invalid coordinate system ({}) or coordinate field ({})".format(
+                        crs_name, coordinate_field
+                    )
+                )
+                raise InvalidCoordSystem
 
-    def getAltitude(self):
+    def _getCoordinateValue(self, value, unit):
         """
-        Get altitude.
+        Return a value of a coordinate variable
+        i) converted to the given unit, if input value has a unit
+        ii) multiplied by the default unit, if input value has no unit assigned
 
-        Returns
-        -------
-        altitude [u.m]
         """
-        return self._altitude * u.m
 
-    @u.quantity_input(altitude=u.m)
-    def setAltitude(self, altitude):
-        """Set altitude."""
-        if None not in [self._altitude]:
-            self._logger.warning("Altitude is already set and will be overwritten")
+        if isinstance(value, u.Quantity):
+            try:
+                return value.to(unit).value
+            except u.UnitsError:
+                self._logger.error("Invalid unit given ({}) for value: {})".format(unit, value))
+                raise
 
-        self._altitude = altitude.value
+        return value
 
-    def hasAltitude(self):
+    def setCoordinates(self, crs_name, xx, yy, zz=None):
         """
-        Return True if tel has altitude.
+        Set coordinates of an array element.
 
-        Returns
-        -------
-        bool
-        """
-        return self._altitude is not None
+        Attributes
+        ----------
+        crs_name: str
+            Name of coordinate system
+        xx: float
+            x-coordinate
+        yy: float
+            y-coordinate
+        zz: float
+            z-coordinate (altitude)
+
+        Raises
+        ------
+        InvalidCoordSystem
+            If coordinate system is not known
 
-    def getMercatorCoordinates(self):
         """
-        Get the latitude and longitude.
+
+        try:
+            self.crs[crs_name]["xx"]["value"] = self._getCoordinateValue(
+                xx, self.crs[crs_name]["xx"]["unit"]
+            )
+            self.crs[crs_name]["yy"]["value"] = self._getCoordinateValue(
+                yy, self.crs[crs_name]["yy"]["unit"]
+            )
+            if zz is not None:
+                self.crs[crs_name]["zz"]["value"] = self._getCoordinateValue(
+                    zz, self.crs[crs_name]["zz"]["unit"]
+                )
+        except KeyError:
+            self._logger.error("Invalid coordinate system ({})".format(crs_name))
+            raise InvalidCoordSystem
+
+    def getAltitude(self):
+        """ "
+        Get altitude of an array element
 
         Returns
         -------
-        (latitude [u.deg], longitude [u.deg])
+        astropy.Quantity
+            telescope altitidue
+
         """
-        return self._latitude * u.deg, self._longitude * u.deg
+        for _crs in self.crs.values():
```

Comment:
- without calling convertAll, only the initial systems are filled 
- after calling convertAll, only those with a given crs are filled (remember the coordinate systems are an argument fo the convert_all function)

e.g., we only might have the mercator system filled, so it would return this one. But if all are filled, it doesn't matter and the first one is returned.

Reply:
OK, I couldn't find a case where we don't call `convertAll`, but better not to rely on it.

Reply:
We always call `convertAll`, but there is no need to provide all coordinate systems in the call `convertAll(self, crsLocal=None, crsWgs84=None, crsUtm=None):`: if only UTM is given, than only one conversion is one.